File: BackEnd/src/main/java/br/com/rd/ecommerce/Model/Entity/Address.java
package br.com.rd.ecommerce.Model.Entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.io.Serializable;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    @Entity
    @Table(name = "tb_endereco")

    public class Address  implements Serializable {


        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id_endereco")
        private Long idEndereco;

        @Column(name = "endereco")
        private String endereco;

        @Column(name = "nro_endereco",length=6)
        private String numero;

        @Column(name = "cep",length=8)
        private String cep;

        @Column(name = "bairro")
        private String bairro;

        @Column(name = "complemento")
        private String complemento;

        @Column(name = "cidade")
        private String cidade;

        @Column(name = "estado")
        private String estado;

        @ManyToOne
        @JoinColumn(name="id_client")
        private Client client;

    }

File: MyIce/src/app/models/Address.ts
export class Address{
    
    constructor(
        public cep:string,
        public logradouro: string,
        public bairro: string,
        public numero:string,
        public uf:string,
        public localidade:string,
        public complemento?:string,
        ){}
    }

//         setEndereco(cep:string, endereco:string, bairro:string, estado:string, cidade:string){
//             this.cep = cep;
//             this._endereco = endereco;
//             this._bairro = bairro;
//             this._estado = estado;
//             this._cidade = cidade;
//         }

//         get cep():string{
//             return this._cep;
//         }
//         set setCep(novoCep: string){
//             this._cep = novoCep;
//         }
        
//         get endereco():string{
//             return this._endereco;
//         }
//         get bairro():string{
//             return this._bairro;
//         }
//         get numero():string{
//             return this._numero;
//         }
//         get estado():string{
//             return this._estado;
//         }
//         get cidade():string{
//             return this._cidade;
//         }

//         get complemento():string{
//             return this._complemento;
//         }
// }